packages/desktop/src/renderer/app/components/menus/databuckets-menu/databuckets-menu.component.ts

```diff
@@ -0,0 +1,175 @@
+import { MainAPI } from 'src/renderer/app/constants/common.constants';
+import { FocusableInputs } from 'src/renderer/app/enums/ui.enum';
+import { ContextMenuEvent } from 'src/renderer/app/models/context-menu.model';
+import { EnvironmentsService } from 'src/renderer/app/services/environments.service';
+import { EventsService } from 'src/renderer/app/services/events.service';
+import { UIService } from 'src/renderer/app/services/ui.service';
+import { updateEnvironmentDatabucketsFilterAction } from 'src/renderer/app/stores/actions';
+import { Store } from 'src/renderer/app/stores/store';
+import { Config } from 'src/shared/config';
+import { Settings } from 'src/shared/models/settings.model';
+
+@Component({
+  selector: 'app-databuckets-menu',
+  templateUrl: './databuckets-menu.component.html',
+  styleUrls: ['./databuckets-menu.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DatabucketsMenuComponent implements OnInit, OnDestroy {
+  @ViewChild('databucketsMenu') private databucketsMenu: ElementRef;
+  public settings$: Observable<Settings>;
+  public activeEnvironment$: Observable<Environment>;
+  public databucketList$: Observable<DataBucket[]>;
+  public activeDatabucket$: Observable<DataBucket>;
+  public databucketsFilter$: Observable<string>;
+  public databucketsFilter: FormControl;
+  public dragIsDisabled = false;
+  public focusableInputs = FocusableInputs;
+  public os$: Observable<string>;
+  public menuSize = Config.defaultRouteMenuSize;
+  private databucketsFilterSubscription: Subscription;
+
+  constructor(
+    private environmentsService: EnvironmentsService,
+    private store: Store,
+    private eventsService: EventsService,
+    private uiService: UIService,
+    private formBuilder: FormBuilder
+  ) {}
+
+  @HostListener('keydown', ['$event'])
+  public escapeFilterInput(event: KeyboardEvent) {
+    if (event.key === 'Escape') {
+      this.clearFilter();
+    }
+  }
+
+  ngOnInit() {
+    this.os$ = from(MainAPI.invoke('APP_GET_OS'));
+    this.databucketsFilter = this.formBuilder.control('');
+
+    this.activeEnvironment$ = this.store.selectActiveEnvironment();
+    this.activeDatabucket$ = this.store.selectActiveDatabucket();
+    this.settings$ = this.store.select('settings');
+    this.databucketsFilter$ = this.store.select('databucketsFilter');
+
+    this.databucketList$ = combineLatest([
+      this.store.selectActiveEnvironment().pipe(
+        filter((activeEnvironment) => !!activeEnvironment),
+        distinctUntilChanged(),
+        map((activeEnvironment) => activeEnvironment.data)
+      ),
+      this.databucketsFilter$.pipe(
+        tap((search) => {
+          this.databucketsFilter.patchValue(search, { emitEvent: false });
+        })
+      )
+    ]).pipe(
+      map(([databuckets, search]) => {
+        this.dragIsDisabled = search.length > 0;
+
+        return databuckets.filter((databucket) =>
+          databucket.name.includes(search)
```

Comment:
add .toLowerCase() to both name and search to be able to search case insensitive